OpenRA.Mods.Common/Widgets/ProductionPaletteWidget.cs

```diff
@@ -334,7 +334,7 @@ bool HandleMiddleClick(ProductionItem item, ProductionIcon icon, int handleCount
 		bool HandleEvent(ProductionIcon icon, MouseButton btn, Modifiers modifiers)
 		{
 			var startCount = modifiers.HasModifier(Modifiers.Shift) ? 5 : 1;
-			var cancelCount = modifiers.HasModifier(Modifiers.Ctrl) ? CurrentQueue.QueueLength : startCount;
+			var cancelCount = modifiers.HasModifier(Modifiers.Ctrl) ? CurrentQueue.AllQueued().ToList().Count : startCount;
```

Comment:
This is worse than the `.Count()`! It forces two enumerations and a copy instead of just one.

In my earlier comment I meant something like:
```csharp
// PERF: We know this is actually a list behind the scenes, so can cheat to avoid an enumeration
var cancelCount = modifiers.HasModifier(Modifiers.Ctrl) ? (List<ProductionItem>)(CurrentQueue.AllQueued()).Count : startCount;
```

Reply:
That doesnt work, im not sure what you mean here. Ill revert to the .Count() for now, but i need some input, as i just dont know how else this could be working, sorry. Il be on the irc for that.